### ISSUE TITLE
When an unselected tab is dragged, narrator does not read the Name and Index of the Tab.

### ISSUE THREAD

**Issue by YuliKl** [microsoft/microsoft-ui-xaml#3507]
**Describe the bug**

**Steps to reproduce the bug**
Pre-requisite:
- Turn on Narrator.
- Set Narrator in Item Mode.

1. Launch XAML Controls Gallery application.
2. Navigate to "All Controls" button and activate it.
3. Navigate to "TabView" and activate it.
4. create at least 3-4 tabs.
5. Navigate to any unselected tab(**Do Not activate this tab item**).
6. Drag the tab using "Alt+Shift+Left/Right arrow" key and observe the narrator reading.

**Expected behavior**

Narrator should read the Name and Index of the dragged tab, also with the current neighboring tab items.

**Screenshots**

![image](https://user-images.githubusercontent.com/13142665/97396087-9c8aa580-18a3-11eb-874a-9adfcbac6941.png)

**Version Info**

Xaml Controls Gallery version 1.2.10.0

**Additional context**

Copied from [internal bug](https://microsoft.visualstudio.com/OS/_queries/edit/23661887)

**Comment by YuliKl** [comment]
Suggestion in the original issue:
For that to happen, you have to appropriately raise PropertyChanged events for structural properties (PositionInSet, for instance) as they change. Right now, tab items in XAML do not.

**Comment by chingucoding** [comment]
Hmm, what currently get's announced is certainly not ideal (something along the lines of "Dropped [item name 1] between [item name 2] and [item name 3]"). The issue however with PropertyChanged is issue #3469 which seems to be an issue with narrator. Would just implementing the property changed be fine, even if narrator currently does not properly handle that? If so, I would be more than happy to work on this.

**Comment by StephenLPeters** [comment]
looking at the internal narrator issue, I think #3469 is specific to numberbox's template setup. I don't think tab view would it that. However if we can't get this working, we shouldn't check in code we cant test.  I'm a bit confused what is going on here though, tabview is just a list view, why is the selected item drag drop different from other items.

**Comment by chingucoding** [comment]
Hmm that's a good point. I've tried adding the code below in the OnItemsChanged of the TabView (since these drag/drop operations are implemented as remove item, then readd item). Narrator didn't seem to pick that up.

```c++
if (const auto peer = winrt::FrameworkElementAutomationPeer::FromElement(*newItem))
{
    auto positionInSet = winrt::AutomationProperties::GetPositionInSet(*newItem);
    winrt::AutomationProperties::SetPositionInSet(*newItem, args.Index());
    peer.RaisePropertyChangedEvent(winrt::AutomationElementIdentifiers::SizeOfSetProperty(),winrt::box_value(positionInSet),winrt::box_value(args.Index()));
}
```

Could it be that the selected item works correctly as it changes the selection and thus should announce the changed selection?
